ldp/alg/callbacks.py

```diff
@@ -139,38 +139,58 @@ async def after_eval_step(self, trajectories: Sequence[Trajectory]) -> None:
 class TrajectoryFileCallback(Callback):
     """Callback that writes trajectories to a file."""
 
-    def __init__(self, output_dir: os.PathLike | str):
+    @staticmethod
+    def default_serialize_env(env: Environment, transition: Transition) -> str | None:
+        """Export a JSON-serialized Frame if the transition is done."""
+        if not transition.done:
+            return None
+        try:
+            frame = env.export_frame()
+        except NotImplementedError:
+            return None  # Allow for envs that didn't implement export_frame()
+        return frame.model_dump_json(exclude={"state"}, indent=2)
```

Comment:
I decided to keep state exclusion for now, because on a test-set run with PaperQA Local, `Frame` exports from just the `done`/`not failed` trajectories takes 0.5 GB. With 10 replicates, ROM usage starts to add up quickly

Reply:
This actually explains why my TS RAM blows up so dramatically - must be something in state.